tests/fixtures/tls_fixtures.go

```diff
@@ -0,0 +1,136 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Reads all markdown files in the specified directory and validates the
+// Ignition configs wrapped in code fences.
+
+package fixtures
+
+// This file contains all the large constants used for testing
+// the tls flow.
+
+var (
+	// PrivateKey is generated via:
+	// openssl ecparam -genkey -name secp384r1 -out server.key
+	PrivateKey = []byte(`-----BEGIN EC PARAMETERS-----
+BgUrgQQAIg==
+-----END EC PARAMETERS-----
+-----BEGIN EC PRIVATE KEY-----
+MIGkAgEBBDB6yW6RIYfTXdYVuPY0V0L6EtZ6vZD86vgbsw52Y3/U5nZ2JE++JrKu
+tt2Xt/NMzG6gBwYFK4EEACKhZANiAAQDEhfHEulYKlANw9eR5l455gwzAIQuraa0
+49RhvM7PPywaiD8DobteQmE8wn7cJSzOYw6GLvrL4Q1BO5EFUXknkW50t8lfnUeH
+veCNsqvm82F1NVevVoExAUhDYmMREa4=
+-----END EC PRIVATE KEY-----`)
+
+	// PublicKey is generated via:
+	// openssl req -new -x509 -sha256 -key server.key -out server.crt -days 3650
+	PublicKey = []byte(`-----BEGIN CERTIFICATE-----
+MIICzTCCAlKgAwIBAgIJALTP0pfNBMzGMAoGCCqGSM49BAMCMIGZMQswCQYDVQQG
+EwJVUzETMBEGA1UECAwKQ2FsaWZvcm5pYTEWMBQGA1UEBwwNU2FuIEZyYW5jaXNj
+bzETMBEGA1UECgwKQ29yZU9TIEluYzEUMBIGA1UECwwLRW5naW5lZXJpbmcxEzAR
+BgNVBAMMCmNvcmVvcy5jb20xHTAbBgkqhkiG9w0BCQEWDm9lbUBjb3Jlb3MuY29t
+MB4XDTE4MDEyNTAwMDczOVoXDTI4MDEyMzAwMDczOVowgZkxCzAJBgNVBAYTAlVT
+MRMwEQYDVQQIDApDYWxpZm9ybmlhMRYwFAYDVQQHDA1TYW4gRnJhbmNpc2NvMRMw
+EQYDVQQKDApDb3JlT1MgSW5jMRQwEgYDVQQLDAtFbmdpbmVlcmluZzETMBEGA1UE
+AwwKY29yZW9zLmNvbTEdMBsGCSqGSIb3DQEJARYOb2VtQGNvcmVvcy5jb20wdjAQ
+BgcqhkjOPQIBBgUrgQQAIgNiAAQDEhfHEulYKlANw9eR5l455gwzAIQuraa049Rh
+vM7PPywaiD8DobteQmE8wn7cJSzOYw6GLvrL4Q1BO5EFUXknkW50t8lfnUeHveCN
+sqvm82F1NVevVoExAUhDYmMREa6jZDBiMA8GA1UdEQQIMAaHBH8AAAEwHQYDVR0O
+BBYEFEbFy0SPiF1YXt+9T3Jig2rNmBtpMB8GA1UdIwQYMBaAFEbFy0SPiF1YXt+9
+T3Jig2rNmBtpMA8GA1UdEwEB/wQFMAMBAf8wCgYIKoZIzj0EAwIDaQAwZgIxAOul
+t3MhI02IONjTDusl2YuCxMgpy2uy0MPkEGUHnUOsxmPSG0gEBCNHyeKVeTaPUwIx
+AKbyaAqbChEy9CvDgyv6qxTYU+eeBImLKS3PH2uW5etc/69V/sDojqpH3hEffsOt
+9g==
+-----END CERTIFICATE-----`)
+
+	// PrivateKey2 is generated via
+	// openssl ecparam -genkey -name secp384r1 -out server.key
+	PrivateKey2 = []byte(`-----BEGIN EC PARAMETERS-----
+BgUrgQQAIg==
+-----END EC PARAMETERS-----
+-----BEGIN EC PRIVATE KEY-----
+MIGkAgEBBDCfXncsl/kqihUWRHThBdGEDpv/bavwHYEi2tjrHiRkm+b7zhFlup8o
+aP1l1zP1LhKgBwYFK4EEACKhZANiAAQ/J0D0C3h2a55JU3/EANe1d3e2/mfcoXGq
+P8soiFdYntRIC4+V4dnRJuHRR+FHR/3531EIf2WXsoIJr/IRhR/j0tAeXpZ++G+E
+vaooXf7gShnhRYKM4viPx4+DhSPjmqw=
+-----END EC PRIVATE KEY-----`)
+
+	// PublicKey2 is generate via:
+	// generate csr:
+	// openssl req -new -key server.key -out server.csr
```

Comment:
Initially, I struggled to generate a `PublicKey` based on the `godoc` provided for other keys, so I thought it would be easier for other people to understand this process if I just jot down the explanation here.

Reply:
These commands from the comments on `PrivateKey`/`PublicKey` worked for me:

```
openssl ecparam -genkey -name secp384r1 -out server.key
openssl req -new -x509 -sha256 -key server.key -out server.crt -days 3650
```

The different commands for the two key pairs implies that there's a reason that they should be generated differently, and I don't think that's the case?

Reply:
To access this key over the local server(127.0.0.1), you need to add the `subjectAltName` in its `X509v3 extensions` spec, so if you use the above two commands, you will end up getting the following result:
```
$ openssl x509 -in server.crt -text -noout
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            24:1a:a5:9b:83:d6:e4:5f:d9:2d:d5:08:25:a0:1f:2d:56:4e:3a:32
        Signature Algorithm: ecdsa-with-SHA256
        Issuer: C = US, ST = NC, L = Raleigh, O = Red Hat, OU = Engineering, CN = coreos
        Validity
            Not Before: May 14 14:00:03 2020 GMT
            Not After : May 12 14:00:03 2030 GMT
        Subject: C = US, ST = NC, L = Raleigh, O = Red Hat, OU = Engineering, CN = coreos
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (384 bit)
                pub:
                    04:8a:79:5b:4a:01:35:e2:a7:98:1f:f4:2f:1b:0c:
                    44:eb:ce:dc:17:5b:8a:2e:52:a1:29:1d:f5:c5:1e:
                    56:22:c3:32:a3:8d:b2:f7:e4:83:ff:bf:dc:60:e1:
                    05:f4:1a:67:1b:c3:be:b1:6b:93:1e:40:11:a0:f9:
                    39:04:44:68:a1:e7:f6:cd:da:d4:cc:2c:1c:4d:01:
                    ef:51:28:cc:1c:3c:37:e0:ba:4e:00:c0:c1:a2:42:
                    bb:d5:12:af:4c:96:83
                ASN1 OID: secp384r1
                NIST CURVE: P-384
        X509v3 extensions:
            X509v3 Subject Key Identifier: 
                E4:1D:F6:C1:83:3C:41:B9:56:53:15:33:2C:F3:7E:5F:A1:06:D9:70
            X509v3 Authority Key Identifier: 
                keyid:E4:1D:F6:C1:83:3C:41:B9:56:53:15:33:2C:F3:7E:5F:A1:06:D9:70

            X509v3 Basic Constraints: critical
                CA:TRUE
    Signature Algorithm: ecdsa-with-SHA256
         30:64:02:30:6f:9b:3f:0a:2c:dc:68:fd:a5:4f:b9:23:76:40:
         95:47:68:76:16:a4:12:20:91:0a:1d:13:93:a6:59:19:b7:08:
         68:41:d2:a1:c5:33:5f:93:3f:2d:30:1f:df:6f:b7:fb:02:30:
         4a:56:bc:ea:2b:8c:ea:72:b5:18:a4:0f:02:2e:ab:bc:5a:10:
         d0:50:dd:c4:5b:58:e4:34:dd:5c:8c:bf:de:04:8e:49:70:09:
         86:e2:69:b2:12:2d:72:0c:51:e4:5f:1b

``` 
However, if you use the instructions written in this `godoc`, you will see the `X509v3 Subject Alternative Name` spec added there in the key:
```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            6c:54:b5:ba:07:04:61:81:90:a1:38:95:e8:ef:ff:af:7c:1d:3b:9f
        Signature Algorithm: ecdsa-with-SHA256
        Issuer: C = US, ST = NC, L = Raleigh, O = Red Hat, OU = Engineering, CN = CoreOS, emailAddress = oem@coreos.com
        Validity
            Not Before: May  7 22:33:07 2020 GMT
            Not After : May  5 22:33:07 2030 GMT
        Subject: C = US, ST = NC, L = Raleigh, O = Red Hat, OU = Engineering, CN = CoreOS, emailAddress = oem@coreos.com
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (384 bit)
                pub:
                    04:3f:27:40:f4:0b:78:76:6b:9e:49:53:7f:c4:00:
                    d7:b5:77:77:b6:fe:67:dc:a1:71:aa:3f:cb:28:88:
                    57:58:9e:d4:48:0b:8f:95:e1:d9:d1:26:e1:d1:47:
                    e1:47:47:fd:f9:df:51:08:7f:65:97:b2:82:09:af:
                    f2:11:85:1f:e3:d2:d0:1e:5e:96:7e:f8:6f:84:bd:
                    aa:28:5d:fe:e0:4a:19:e1:45:82:8c:e2:f8:8f:c7:
                    8f:83:85:23:e3:9a:ac
                ASN1 OID: secp384r1
                NIST CURVE: P-384
        X509v3 extensions:
            X509v3 Subject Alternative Name: 
                IP Address:127.0.0.1
            X509v3 Subject Key Identifier: 
                A2:F5:60:58:D1:45:3E:1A:C5:ED:7C:DA:46:D7:83:9E:97:CF:5F:1A
            X509v3 Authority Key Identifier: 
                keyid:A2:F5:60:58:D1:45:3E:1A:C5:ED:7C:DA:46:D7:83:9E:97:CF:5F:1A

            X509v3 Basic Constraints: critical
                CA:TRUE
    Signature Algorithm: ecdsa-with-SHA256
         30:64:02:30:6f:08:8a:fd:93:8d:e8:47:5f:19:e0:72:d3:cd:
         a4:5f:cd:a7:05:ce:8e:f9:2e:de:67:d2:99:37:3c:0e:42:ad:
         48:37:b1:75:3a:a3:82:31:c8:b3:45:1f:32:09:9f:b7:02:30:
         64:cc:54:a1:33:a1:26:43:72:5c:64:12:0b:b9:dd:63:3b:3d:
         09:63:1e:02:fd:14:94:9c:12:de:21:cc:ed:ae:14:53:63:86:
         ad:0e:0d:5d:4b:35:cd:62:f2:94:02:02
```
I'm pretty sure that `openssl req -new -x509 -sha256 -key server.key -out server.crt -days 3650` command can be modified to add `subjectAltName` field in its spec, but I didn't see any good documentation to address that concern.

Reply:
Whoops, I missed `subjectAltName`, thanks.  Could you update the other comment as well?

Reply:
Done!

Reply:
I'm not seeing this in your latest push.

Reply:
As requested, please update the comment on `PublicKey`.